.github/workflows/neuron-ci.yml

```diff
@@ -0,0 +1,245 @@
+    
+    timeout-minutes: 30
+        
+    env:
+      INSTALL_DIR: install
+      SDK_ROOT: $(xcrun --sdk macosx --show-sdk-path)
+      TRAVIS: true
+
+    strategy:
+      matrix:
+        os: [ ubuntu-18.04, macOS-10.15 ]
+        config:
+          - { matrix_eval : "CC=gcc CXX=g++" ,      build_mode: "cmake",      cmake_option: "-DNRN_ENABLE_BINARY_SPECIAL=ON -DNRN_ENABLE_MPI=ON -DNRN_ENABLE_INTERVIEWS=ON -DNRN_ENABLE_CORENEURON=ON", documentation: ON }
+          - { matrix_eval : "CC=gcc-8 CXX=g++-8",   build_mode: "setuptools"  }
+          - { matrix_eval : "CC=gcc CXX=g++",       build_mode: "autotools",  config_options: "--without-x --without-paranrn --with-nrnpython=$PYTHON2 PYTHON_BLD=$PYTHON2", use_python2 : ON }
+          - { matrix_eval : "CC=gcc-8 CXX=g++-8",   build_mode: "autotools",  config_options: "--without-x --with-paranrn=dynamic --with-nrnpython=$PYTHON" }
+          - { matrix_eval : "CC=gcc-5 CXX=g++-5",   build_mode: "cmake",      cmake_option: "-DNRN_ENABLE_MPI=OFF -DNRN_ENABLE_INTERVIEWS=OFF -DNRN_ENABLE_CORENEURON=ON" }
+          - { matrix_eval : "CC=gcc-6 CXX=g++-6",   build_mode: "cmake",      cmake_option: "-DNRN_ENABLE_MPI=ON -DNRN_ENABLE_INTERVIEWS=ON -DNRN_ENABLE_CORENEURON=OFF" }
+          - { matrix_eval : "CC=gcc-9 CXX=g++-9",   build_mode: "cmake",      cmake_option: "-DNRN_ENABLE_PYTHON=OFF -DNRN_ENABLE_RX3D=OFF -DNRN_ENABLE_CORENEURON=ON" }
+          - { matrix_eval : "CC=gcc-8 CXX=g++-8",   build_mode: "cmake",      python_dynamic : ON }
+      fail-fast: false
+      
+    steps:
+      - name: Setup cmake
+        uses: jwlawson/actions-setup-cmake@v1.4
+        with:
+          cmake-version: ${{ env.DESIRED_CMAKE_VERSION }}
+
+      - name: Install homebrew packages
+        if: startsWith(matrix.os, 'macOS')
+        run: |
+          brew install coreutils doxygen mpich xz autoconf automake libtool && brew unlink mpich && brew install openmpi && brew cask install xquartz
+        shell: bash
+        
+      - name: Install apt packages
+        if: startsWith(matrix.os, 'ubuntu')
+        run: |
+          sudo apt-get install  build-essential doxygen autoconf automake libtool autotools-dev libopenmpi-dev libmpich-dev libx11-dev libxcomposite-dev mpich openmpi-bin patchelf python-pip python-tk python-numpy python3-numpy python3-pip g++-5 g++-6 g++-8 g++-9  
+        shell: bash
+
+      - name: Set up Python2
+        uses: actions/setup-python@v2
+        with:
+          python-version: '2.x'
+
+      - name: Install Python2 dependencies
+        if: ${{ matrix.config.use_python2 == 'ON' || matrix.config.python_dynamic == 'ON' }}
+        run: python2 -m pip install --upgrade pip wheel setuptools scikit-build pytest matplotlib bokeh ipython cython pytest mpi4py
+
+      - name: Set up Python3
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
```

Comment:
Does it matter here if we don't check if `use_python2 != 'ON'`?

Reply:
Will check

Reply:
It's needed for python dynamic feature